codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ActionTestKitGenerator.scala

```diff
@@ -135,8 +140,9 @@ object ActionTestKitGenerator {
     service.commands
       .map { command =>
         s"""|public ${selectOutputResult(command)} ${lowerFirst(command.name)}(${selectInputType(command)} ${lowerFirst(
-          command.inputType.protoName)}) {
-          |  ${selectOutputEffect(command)} effect = createAction().${lowerFirst(command.name)}(${lowerFirst(
+          command.inputType.protoName)}, Metadata metadata, Optional<String> eventSubject) {
```

Comment:
In "production" the "event subject" comes from the metadata entry `ce-subject`, I wonder if we should extract that like that from user provided Metadata here instead of having a specific parameter for actions?

And maybe no default if it is not set, as in a normal call to an action? I think it may be set for topic in, and always for eventing in, so for testing such actions a user can specify it through the passed Metadata.

Reply:
I'm happy to hear that the event subject comes from the metadata in prod. It just didn't look good that parameter constructor. Cool, I'll extract if there's a key `ce-subject` otherwise won't be set. Meaning `TestKitActionContext.eventSubject()` would return `None` (or `Optional.empty()`) when no `ce-subject` in its metadata.

Reply:
Yep, makes sense. For entities it will be a bit different since they _are_ the subject and it will never change for the same one, I don't think they will ever see a `ce-subject` in the metadata, and if they do it does not imply the same thing as for an action.